projects/plugins/jetpack/modules/carousel/jetpack-carousel.js

```diff
@@ -1546,41 +1117,58 @@
 					// Initially, the image is a 1x1 transparent gif.
 					// The preview is shown as a background image on the slide itself.
 					var image = new Image();
-					image.src =
-						'data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7';
-					image.style.width = '100%';
-					image.style.height = '100%';
+					image.src = attrs.src;
+					// image.style.width = '100%';
+					// image.style.height = '100%';
 
 					var slideEl = document.createElement( 'div' );
-					slideEl.classList.add( 'jp-carousel-slide' );
+					slideEl.classList.add( 'swiper-slide' );
 					slideEl.setAttribute( 'itemprop', 'associatedMedia' );
 					slideEl.setAttribute( 'itemscope', '' );
 					slideEl.setAttribute( 'itemtype', 'https://schema.org/ImageObject' );
-					domUtil.hide( slideEl );
+					// domUtil.hide( slideEl );
+					var zoomEl = document.createElement( 'div' );
+					zoomEl.classList.add( 'swiper-zoom-container' );
 
-					slideEl.style.left = i < startIndex ? -1000 : carousel.gallery.offsetWidth;
+					// slideEl.style.left = i < startIndex ? -1000 : carousel.gallery.offsetWidth;
 					carousel.gallery.appendChild( slideEl );
-					slideEl.appendChild( image );
 
+					slideEl.appendChild( zoomEl );
+					zoomEl.appendChild( image );
 					slideEl.setAttribute( 'data-attachment-id', attrs.attachmentId );
 					slideEl.setAttribute( 'data-permalink', attrs.permalink );
 					slideEl.setAttribute( 'data-orig-file', attrs.origFile );
 
 					if ( useInPageThumbnails ) {
 						// Use the image already loaded in the gallery as a preview.
 						attrs.previewImage = attrs.src;
-						slideEl.style.backgroundImage = 'url("' + attrs.src + '")';
-						slideEl.style.backgroundSize = '100% 100%';
-						slideEl.style.backgroundPosition = 'center center';
 					}
 
 					var slide = { el: slideEl, attrs: attrs, index: i };
 					carousel.slides.push( slide );
-					fitSlides( [ slide ] );
 				}
 			} );
 		}
 
+		function loadSwiper( gallery, options ) {
+			if ( ! window.Swiper ) {
+				var loader = document.querySelector( '#jp-carousel-loading-overlay' );
+				domUtil.show( loader );
+				var head = document.getElementsByTagName( 'head' )[ 0 ];
+				var jsScript = document.createElement( 'script' );
+				jsScript.id = 'jetpack-carousel-swiper-js';
```

Comment:
This is fine, but I just wanted to point out that it's not an issue to have the script tag without an ID 👍

Reply:
I was tossing up whether to checking for the existing of this element via id before adding to avoid appending more than once, but looking at the code path I think the check for Swiper object should prevent this from happening - but will do some more testing today with double clicking, etc. to make sure there isn't any race condition that could cause a double appending of the script, if not I will remove the id.